docs/selfhosting/configure-infra.md

```diff
@@ -0,0 +1,429 @@
+---
+title: Configure Infrastructure
+seotitle: Configure Infrastructure
+seodesc: Learn how to configure infrastructure resources for your Encore app.
+---
+
+If you are using infrastructure resources, such as SQL databases, Pub/Sub, or metrics, you will need to configure your Docker image with the necessary configuration.
+The `build` command lets you provide this by specifying a path to a config file using the `--config` flag.
+
+```bash
+encore build docker --config path/to/infra-config.json MY-IMAGE:TAG
+```
+
+The configuration file should be a JSON file using the [Encore Infra Config](https://encore.dev/schemas/infra.schema.json) schema.
+
+This includes configuring things like:
+
+- How to access infrastructure resources (what provider to use, what credentials to use, etc.)
+- How to call other services over the network ("service discovery"),
+  most notably their base URLs.
+- Observability configuration (where to export metrics, etc.)
+- Metadata about the environment the application is running in, to power Encore's [metadata APIs](/docs/develop/metadata).
+- The values for any application-defined secrets.
+
+This configuration is necessary for the application to behave correctly.
+
+## Example
+
+Here's an example configuration file you can use.
+
+```json
+{
+  "$schema": "https://encore.dev/schemas/infra.schema.json",
+  "metadata": {
+    "app_id": "my-app",
+    "env_name": "my-env",
+    "env_type": "production",
+    "cloud": "gcp",
+    "base_url": "https://my-app.com"
+  },
+  "sql_servers": [
+    {
+      "host": "my-db-host:5432",
+      "databases": {
+        "my-db": {
+          "username": "my-db-owner",
+          "password": {"$env": "DB_PASSWORD"}
+        }
+      }
+    }
+  ],
+  "service_discovery": {
+    "myservice": {
+      "base_url": "https://my-service:8044"
```

Comment:
Probably better to use the same hostname as the service name, since that's the most common setup?

```suggestion
      "base_url": "https://myservice:8044"
```